packages/mermaid/src/styles.ts

```diff
@@ -73,8 +73,13 @@ const getStyles = (
 `;
 };
 
-export const addStylesForDiagram = (type: string, diagramTheme: unknown): void => {
-  themes[type] = diagramTheme;
+export const addStylesForDiagram = (
+  type: string,
+  diagramTheme?: (options?: any) => string
+): void => {
+  if (diagramTheme !== undefined) {
+    themes[type] = (options) => diagramTheme(options);
```

Comment:
```suggestion
    themes[type] = diagramTheme;
```

Can you try if this works?

Reply:
Yes, it works.

Fixed.